### ISSUE TITLE
[Bug] PDF Viewer Skips or Shuffles Pages

### ISSUE THREAD

**Issue by OkyDooky** [SimpleMobileTools/Simple-File-Manager#663]
With this latest update, the viewer will sometimes skip a page or randomly swap one page with another. For instance, if I'm scrolling several pages down, page 4 may display the first page. And then scrolling up to page 1 may display page 7 in its place.
This kind of thing happens randomly/inconsistently, but often. One thing that seems to exasperate it and increase the likelihood of the next page, or the one after, being incorrect is zooming in and out of the current page for any decent amount of time.

The device this was first noticed on is a Moto G7 Supra (Power rebrand) running Android 9. I have one other device and I'll test to see if it has similar issues.

**Comment by tibbi** [comment]
does it happen to huge files only?

**Comment by OkyDooky** [comment]
No, apparently. I managed to get it to misbehave with a <1MB PDF.
![251215](https://user-images.githubusercontent.com/59938024/208231591-8b15a63e-d873-4860-90f6-69f967610da0.png)
You can see the page number at the bottom of the displayed document does not match the viewer's number in two of the pages in this screenshot.
[250855.pdf](https://github.com/SimpleMobileTools/Simple-File-Manager/files/10251201/250855.pdf)

**Comment by tibbi** [comment]
works just fine to me, I dont see how would it break. How are you opening the file, through our File Manager or some other intents from other apps?

**Comment by OkyDooky** [comment]
I am opening it direct from and through the app.
The error is not consistent, meaning it will not shuffle incorrectly the same way each time. I tried to force the error by spreading to zoom in on each page for some time, panning across it a bunch then tapping until zoomed back out, before moving to the next page and repeating this until I saw a mismatch.
The mismatch can be "resolved" by sufficiently scrolling far enough away and then back. There doesn't seem to be a fixed amount, but four or five pages seems to be a decent average. In other words, if I scroll to page 4 and it displays page 1, I could get it to display the proper page by scrolling several pages or more down and then back up to 4. Again, that's just a non-specific example, because it seems to occur randomly, or at least inconsistently.
